pkg/sdk/testint/secrets_gen_integration_test.go

```diff
@@ -0,0 +1,673 @@
+package testint
+
+import (
+	"database/sql"
+	"testing"
+	"time"
+
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/acceptance/helpers/random"
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/sdk"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+const Day = 24 * time.Hour
+
+func TestInt_Secrets(t *testing.T) {
+	client := testClient(t)
+	ctx := testContext(t)
+
+	integrationId := testClientHelper().Ids.RandomAccountObjectIdentifier()
+
+	refreshTokenExpiryTime := time.Now().Add(Day).Format(time.DateOnly)
+
+	stringDateToSnowflakeTimeFormat := func(inputLayout, date string) *time.Time {
+		parsedTime, err := time.Parse(inputLayout, date)
+		require.NoError(t, err)
+
+		loc, err := time.LoadLocation("America/Los_Angeles")
+		require.NoError(t, err)
+
+		adjustedTime := parsedTime.In(loc)
+		return &adjustedTime
+	}
+
+	cleanupIntegration := func(t *testing.T, integrationId sdk.AccountObjectIdentifier) func() {
+		t.Helper()
+		return func() {
+			err := client.SecurityIntegrations.Drop(ctx, sdk.NewDropSecurityIntegrationRequest(integrationId).WithIfExists(true))
+			require.NoError(t, err)
+		}
+	}
+
+	err := client.SecurityIntegrations.CreateApiAuthenticationWithClientCredentialsFlow(
+		ctx,
+		sdk.NewCreateApiAuthenticationWithClientCredentialsFlowSecurityIntegrationRequest(integrationId, true, "foo", "foo").
+			WithOauthAllowedScopes([]sdk.AllowedScope{{Scope: "foo"}, {Scope: "bar"}}),
+	)
+	require.NoError(t, err)
+	t.Cleanup(cleanupIntegration(t, integrationId))
+
+	cleanupSecret := func(t *testing.T, id sdk.SchemaObjectIdentifier) func() {
+		t.Helper()
+		return func() {
+			err := client.Secrets.Drop(ctx, sdk.NewDropSecretRequest(id).WithIfExists(true))
+			require.NoError(t, err)
+		}
+	}
+
+	createSecretWithOAuthClientCredentialsFlow := func(t *testing.T, integrationId sdk.AccountObjectIdentifier, scopes []sdk.SecurityIntegrationScope, with func(*sdk.CreateWithOAuthClientCredentialsFlowSecretRequest)) (*sdk.Secret, sdk.SchemaObjectIdentifier) {
+		t.Helper()
+		id := testClientHelper().Ids.RandomSchemaObjectIdentifier()
+		request := sdk.NewCreateWithOAuthClientCredentialsFlowSecretRequest(id, integrationId, scopes)
+		if with != nil {
+			with(request)
+		}
+		err := client.Secrets.CreateWithOAuthClientCredentialsFlow(ctx, request)
+		require.NoError(t, err)
+		t.Cleanup(cleanupSecret(t, id))
+
+		secret, err := client.Secrets.ShowByID(ctx, id)
+		require.NoError(t, err)
+
+		return secret, id
+	}
```

Comment:
Now, instead of defining these functions here, we add them to pkg/acceptance/helpers/*_client.go